pkg/postage/service.go

```diff
@@ -44,27 +42,29 @@ type Service interface {
 // stores the active batches.
 type service struct {
 	lock         sync.Mutex
-	store        storage.StateStorer
+	store        storage.Store
 	postageStore Storer
 	chainID      int64
 	issuers      []*StampIssuer
```

Comment:
We should remove this in-mem state of StampIssuer completely. The problem is due to the nature of the operations in API, if we dont commit the transaction to storer, we should not be updating the stamp indexes into stamper store. The `save` function is only called on successful completion of the upload.

In order to have proper synchronization, every time we do a GetStampIssuer, we should read the state from the stamper store. Also, all other updates like expiry etc should also be done on the store directly. Also, we should only allow access to a particular stampIssuer to 1 routine as multiple routines can try to use the same indexes if this is no longer in memory. This hit in performance is okay as we only call GetStampIssuer once during upload.